ouroboros-network/src/Ouroboros/Network/Protocol/KeepAlive/Codec.hs

```diff
@@ -0,0 +1,55 @@
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE GADTs               #-}
+{-# LANGUAGE KindSignatures      #-}
+{-# LANGUAGE PolyKinds           #-}
+{-# LANGUAGE RankNTypes          #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Ouroboros.Network.Protocol.KeepAlive.Codec
+  ( codecKeepAlive
+  ) where
+
+import           Control.Monad.Class.MonadST
+
+import           Data.ByteString.Lazy (ByteString)
+
+import qualified Codec.CBOR.Encoding as CBOR (Encoding, encodeWord)
+import qualified Codec.CBOR.Read     as CBOR
+import qualified Codec.CBOR.Decoding as CBOR (Decoder, decodeWord)
+
+import           Network.TypedProtocol.Core
+
+import           Ouroboros.Network.Codec
+import           Ouroboros.Network.Protocol.KeepAlive.Type
+
+
```

Comment:
You should define `ProtocolTimeLimits` and `ProtocolSizeLimits` for the protocol.

Reply:
I left this, together with an implementation of a client / server for another RP.